Clean up README

rgwood opened the pull request:
# Description

The README was getting very long and some places were out of date. I've done a pass to update out-of-date info and remove some info that the book covers better.

fdncred commented:
thanks. it's been needing this pass through for a while. although, I think I'd vote to keep the installation section.

sholderbach commented on README.md:
````diff
@@ -155,72 +107,49 @@ For example, you can load a .toml file as structured data and explore it:
-To set one of these variables, you can use `config set`. For example:
-
-```shell
-> config set line_editor.edit_mode "vi"
-> config set path $nu.path
````
Wow yeah that is out of date

sholderbach commented on README.md:
```diff
@@ -16,78 +16,28 @@ A new type of shell.
-We also have an active [Discord](https://discord.gg/NtAbbGn) and [Twitter](https://twitter.com/nu_shell) if you'd like to come and chat with us.
 
-You can also find information on more specific topics in our [cookbook](https://www.nushell.sh/cookbook/).
+If you're a developer who would like to contribute to Nu, we're working on a [book for developers](https://www.nushell.sh/contributor-book/) to help you get started. There are also [good first issues](https://github.com/nushell/nushell/issues?q=is%3Aopen+is%3Aissue+label%3A%22good+first+issue%22) to help you dive in.
```
I think the Contributor book is not yet updated and thus this leads to a stub.

sholderbach reviewed (commented):
One suggestion maybe add a TOC at the top to quickly navigate to either pages in the same readme or to the appropriate page like I did it for reedline. This way, we can keep the full installation information (if up to date) there but can keep the easy and interesting to read stuff at the top.

rgwood commented:
Thanks for the feedback; things got a bit busy over here but I’ll do another revision as soon as I have time.

rgwood commented:
OK, did another pass. Notable changes:

- Updated `CONTRIBUTING.md`, it was very out of date
- Added some quick-install info back to the installation section
- Added a table of contents (the [Markdown All in One](https://marketplace.visualstudio.com/items?itemName=yzhang.markdown-all-in-one) extension is nice for this!)
- Removed mention of the contributor book since it's currently a work in progress

It occurrect to me that the Philosophy section could use a rewrite; it's currently a bit crowded, has lots of info that doesn't really fit under "Philosophy". But I'll leave that for another time.

fdncred commented on README.md:
```diff
@@ -8,144 +7,112 @@
+# Linux and macOS
+brew install nushell
+# Windows
+winget install nu
```
This conflicts what the documentation for winget https://www.nushell.sh/book/installation.html#package-managers

I'm not sure which is right.

rgwood replied:
`Nushell.Nushell` is the unambiguous ID for Nushell on winget, probably easiest to use that. Fixed.
<img width="489" alt="image" src="https://user-images.githubusercontent.com/26268125/172462965-4f41c71f-eb46-4350-88f4-a716ed059893.png">

rgwood replied:
I'll update the website in a bit.

rgwood replied:
On second thought.... `winget install nushell` works and it's easier to memorize. Will go with that.

fdncred commented on README.md:
```diff
@@ -270,20 +210,15 @@ Nu is in heavy development and will naturally change as it matures and people us
 
 Please submit an issue or PR to be added to this list.
 
-###   Integrations
```
Why are we deleting the integrations section?

rgwood replied:
All the info's still there under the "Officially Supported By" header; I just thought it was unnecessary to have both "Officially Supported By" and "Integrations" headers.
